### PR TITLE
Modin GS and XGBoost GS Samples, Removed Python2.7, renamed daal4py samples and related files

### PR DESCRIPTION
**Pull request by raoberman**
# Description

Added Intel Modin GS and IntelPython_XGBoost GS samples and related files, removed Python2.7 dependency (request), renamed daal4py samples and related files

Fixes # internal email issue

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Sample Migration (Moving sample from old repository after completing checklist established)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode
- [X] DevCloud

# Checklist for Moving samples:
Links and Details can be found in the samples WG Teams Files. 

- [X] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts 
- [X] Implement coding guidelines and ensure code quality.
- [X] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [X] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [X] Adhere to readme template 
- [X] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [X] Enforce format via clang-format config file
- [X] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [In progress] Review readme with Tom Lenth or Joe Oster. (GitHub User: JoeOster)
- [X] Tested using Dev Cloud when applicable
- [X] Implement fixes for ONSAM Jiras
- [X] If you have new dependencies/binaries, inform Samples Project Manager Swapna R Dontharaju (@srdontha) or @JoeOster

### CODE REVIEW

**Comment by raoberman** [comment]
@JoeOster please note Python 2.7 removal on README.

**Review by JoeOster — DISMISSED** [review]

**Review by anjgola — APPROVED** [review]
Run tests.

**Review by JoeOster — APPROVED** [review]

**Comment by raoberman** [comment]
@JoeOster (or Tom) please review Intel_Modin_GS and IntelPython_XGBoost_GS sample README.mds in Frameworks folder

File: Libraries/oneDAL/IntelPython_daal4py_Distributed_LinearRegression/License.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Copyright Intel Corporation
```
Comment (JoeOster) [inline Libraries/oneDAL/IntelPython_daal4py_Distributed_LinearRegression/License.txt:1]: Please change the text of the license file to the one [here](https://github.com/oneapi-src/oneAPI-samples/blob/master/License.txt)

File: Libraries/oneDAL/IntelPython_daal4py_Getting_Started/License.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Copyright Intel Corporation
+
```
Comment (JoeOster) [inline Libraries/oneDAL/IntelPython_daal4py_Getting_Started/License.txt:2]: Please change the text of the license file to the one [here](https://github.com/oneapi-src/oneAPI-samples/blob/master/License.txt)

File: Libraries/oneDAL/IntelPython_daal4py_Distributed_LinearRegression/README.md [hunk 3]
```diff
@@ -1,11 +1,11 @@
-# daal4py Distributed Linear Regression
```
Comment (JoeOster) [inline Libraries/oneDAL/IntelPython_daal4py_Distributed_LinearRegression/README.md:1]: Old styleReadme, please update using the [mandelbrot readme](https://github.com/oneapi-src/oneAPI-samples/blob/master/DirectProgramming/DPC%2B%2B/CombinationalLogic/mandelbrot/README.md) as an example: In this case a "Purpose" and "License" sections are missing at a minimum

File: Libraries/oneDAL/IntelPython_daal4py_Distributed_Kmeans/README.md [hunk 4]
```diff
@@ -1,11 +1,11 @@
-# daal4py Distributed K-Means
```
Comment (JoeOster) [inline Libraries/oneDAL/IntelPython_daal4py_Distributed_Kmeans/README.md:1]: Old styleReadme, please update using the [mandelbrot readme](https://github.com/oneapi-src/oneAPI-samples/blob/master/DirectProgramming/DPC%2B%2B/CombinationalLogic/mandelbrot/README.md) as an example: In this case a "Purpose" and "License" sections are missing at a minimum

File: Frameworks/IntelPython_XGBoost/IntelPython_XGBoost_GettingStarted/README.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
+# Intel Python XGBoost Getting Started
```
Comment (JoeOster) [inline Frameworks/IntelPython_XGBoost/IntelPython_XGBoost_GettingStarted/README.md:1]: Old styleReadme, please update using the [mandelbrot readme](https://github.com/oneapi-src/oneAPI-samples/blob/master/DirectProgramming/DPC%2B%2B/CombinationalLogic/mandelbrot/README.md) as an example: In this case a "Purpose" and "License" sections are missing at a minimum

**Review by JoeOster — CHANGES_REQUESTED** [review]

File: Libraries/oneDAL/IntelPython_daal4py_Getting_Started/README.md [hunk 6]
```diff
@@ -1,11 +1,11 @@
-# daal4py Getting Started
```
[the remark this answers is not in the document]
Comment (raoberman) [reply Libraries/oneDAL/IntelPython_daal4py_Getting_Started/README.md:1]: Joe, Not sure which ReadMe you are looking at (daal4py sample renamed). This was fixed in beta09 PR, seenew README which has purpose and license: https://github.com/raoberman/oneAPI-samples/tree/master/Libraries/oneDAL/IntelPython_daal4py_Getting_Started

**Review by JoeOster — DISMISSED** [review]

**Comment by tomlenth** [comment]
In XGBoost,  the image in Running the Sample says "Modin_GettingStarted".  should it be XGBoost?

**Comment by raoberman** [comment]
@tomlenth Modin GettingStarted image is used just as example (also says so in README)- same logic applies for XGBoost sample

**Review by JoeOster — DISMISSED** [review]

**Review by tomlenth — APPROVED** [review]

**Comment by raoberman** [comment]
@anjgola @JoeOster  the Modin test continues to fail because of no current environments have Modin installed. I requested from CI team to make an environment with Modin, along with env creation instructions, but have no update that this has been made

**Comment by raoberman** [comment]
@anjgola @JoeOster  please run test again, added env creation instructions

**Review by JoeOster — DISMISSED** [review]
Kicking off ci again

**Comment by raoberman** [comment]
@JoeOster @anjgola please initiate ci again, i modified the install instructions even though not sure why it failed (ran perfectly fine on my machine and DevCloud fresh install).

**Comment by raoberman** [comment]
@JoeOster @anjgola please initiate ci again, i modified the install instructions even though not sure why it failed (ran perfectly fine on my machine and DevCloud fresh install).

**Comment by JoeOster** [comment]
> @JoeOster @anjgola please initiate ci again, i modified the install instructions even though not sure why it failed (ran perfectly fine on my machine and DevCloud fresh install).

I have requested it to be run again

**Comment by raoberman** [comment]
> > @JoeOster @anjgola please initiate ci again, i modified the install instructions even though not sure why it failed (ran perfectly fine on my machine and DevCloud fresh install).
> 
> I have requested it to be run again

Thank you @JoeOster for your help, much appreciated :)

**Review by anjgola — APPROVED** [review]

**Review by JoeOster — APPROVED** [review]
approving for ci

**Review by JoeOster — APPROVED** [review]
approving for ci

**Comment by raoberman** [comment]
@malikabhi @sravanikonda  please approve to reinitiate CI validation. Thank you

**Review by JoeOster — APPROVED** [review]

**Comment by raoberman** [comment]
@JoeOster fixed the Modin sample.json issue and validated in JSONLint. please initiate CI validation (hopefully last time :) ) thanks in advanced

**Review by JoeOster — APPROVED** [review]

**Comment by raoberman** [comment]
@JoeOster @malikabhi @sravanikonda CI tests passed! please approve merge.  thanks in advanced

**Pull request merged**